python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition_evaluator.py

```diff
@@ -114,12 +115,17 @@ def prefetch(self) -> None:
         self.logger.info("Done prefetching asset records.")
 
     def evaluate(self) -> Tuple[Sequence[AutomationResult], Sequence[EntitySubset[EntityKey]]]:
+        return asyncio.run(self.async_evaluate())
+
+    async def async_evaluate(
+        self,
+    ) -> Tuple[Sequence[AutomationResult], Sequence[EntitySubset[EntityKey]]]:
         self.prefetch()
         num_conditions = len(self.entity_keys)
         num_evaluated = 0
-        for entity_key in self.asset_graph.toposorted_entity_keys:
-            if entity_key not in self.entity_keys:
-                continue
+
+        async def _evaluate_entity_async(entity_key: EntityKey) -> None:
+            nonlocal num_evaluated
```

Comment:
nit: can we just have this function return int instead and sum over the gathered coroutines